compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -428,12 +428,33 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                     }
                 }
 
-                err.span_suggestion(
-                    call_span,
-                    &format!("try calling `{}` as a method", ident),
-                    format!("self.{}({})", path_str, args_snippet),
-                    Applicability::MachineApplicable,
-                );
+                if source.is_call() {
+                    if let Some(val) = res {
```

Comment:
```suggestion
                    if let Some(res) = res.filter(|_| source.is_call()) {
```
to not duplicate the else branch